.github/workflows/test.yml

```diff
@@ -21,15 +21,16 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
+        id: setup-node
         with:
-          node-version: '16'
+          node-version-file: 'package.json'
           check-latest: true
       - uses: actions/cache@v3
         with:
           path: |
             node_modules
             */*/node_modules
-          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock', 'patches/*.patch') }}
+          key: ${{ runner.os }}-node-${{ steps.setup-node.outputs.node-version }}-${{ hashFiles('**/yarn.lock', 'patches/*.patch') }}
```

Comment:
I initially got issues on the PR, because it was using cached `node_modules` produced with a different version of node. This ensures the cache depends on the node version used to avoid unexpected issues.